### PR TITLE
Add eu.cichy1173.tabela

### PR DESCRIPTION
**Pull request by cichy1173**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: eu.cichy1173.tabela.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+app-id: eu.cichy1173.tabela
+runtime: org.kde.Platform
+runtime-version: '6.2'
+sdk: org.kde.Sdk
+command: tabela
+build-options:
+ build-args:
+  - --share=network
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:8]: this is not allowed on our builders. build must be done offline.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
please try to build locally at least once.

File: eu.cichy1173.tabela.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+app-id: eu.cichy1173.tabela
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:1]: `cichy1173.eu` seems to be a parked domain or something.

File: eu.cichy1173.tabela.yml (hunk 1 above)
Comment (cichy1173) [reply eu.cichy1173.tabela.yml:8]: Hi. Thank You for your response. Can you tell me how to fix this? I am sorry, but I am new in this. 

I built this locally from dir and using codeberg repository, but for building fro repo, this build-arg was needed.

File: eu.cichy1173.tabela.yml (hunk 2 above)
Comment (cichy1173) [reply eu.cichy1173.tabela.yml:1]: Yes, This is my domain :-). Check about.cichy1173.eu :-)

**Comment by hfiguiere** [comment]
bot, build eu.cichy1173.tabela

File: eu.cichy1173.tabela.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+runtime-version: '6.2'
+sdk: org.kde.Sdk
+command: tabela
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --filesystem=host
+desktop-file-name: eu.cichy1173.eu.desktop
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:12]: ```(flatpak-builder:2): flatpak-builder-WARNING **: 13:01:38.950: Unknown property desktop-file-name for type BuilderManifest```

File: eu.cichy1173.tabela.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+modules:
+  - name: tabela
+    buildsystem: simple
+    build-commands:
+      - python -m pip install --prefix=/app .
+    sources:
+      - type: git
+        url: https://codeberg.org/cichy1173/tabela-flatpak.git
+        commit: 73753ba98060e0b8f792f306e36eafbe941b6b55
+    build-extensions:
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:22]: (flatpak-builder:2): flatpak-builder-WARNING **: 13:01:38.950: Unknown property build-extensions for type BuilderModule

File: eu.cichy1173.tabela.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      - org.freedesktop.Sdk.Extension.python3.9
+    extra-data:
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:24]: `(flatpak-builder:2): flatpak-builder-WARNING **: 13:01:38.950: Unknown property extra-data for type BuilderModule`

**Review by hfiguiere — CHANGES_REQUESTED** [review]
These there error message are why I asked if you tried to build locally. Obviously you ignored them.

Also ou need to add the python dependencies for the build to work.

See for tools that can help
https://github.com/flatpak/flatpak-builder-tools/tree/master/pip

**Comment by cichy1173** [comment]
> These there error message are why I asked if you tried to build locally. Obviously you ignored them.

Sorry, I managed to build on this manifest once. It is possible that it was a bug or some bug in GNOME Builder. Local build is now working. 

Thanks for your work and sorry for the faulty PR. I am new to this.

**Comment by hfiguiere** [comment]
bot, build eu.cichy1173.tabela

**Comment by hfiguiere** [comment]
The mandatory appstream file is not found. I don't see one being installed.

**Comment by hfiguiere** [comment]
Nor is an icon or a desktop file.

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

File: eu.cichy1173.tabela.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    build-commands:
+      - python -m pip install --no-deps --no-build-isolation --prefix=/app .
+      - mkdir -p /app/share/applications
+      - cp data/eu.cichy1173.tabela.desktop /app/share/applications
+      - mkdir -p /app/share/metainfo
+      - cp data/eu.cichy1173.tabela.metainfo.xml /app/share/metainfo
+      - mkdir -p /app/share/icons/hicolor/48x48/apps
+      - cp data/icons/hicolor/apps/48x48/eu.cichy1173.tabela.png /app/share/icons/hicolor/48x48/apps
+      - mkdir -p /app/share/icons/hicolor/128x128/apps
+      - cp data/icons/hicolor/apps/128x128/eu.cichy1173.tabela.png /app/share/icons/hicolor/48x48/apps
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:60]:
```suggestion
      - install -Dm644 data/eu.cichy1173.tabela.desktop -t /app/share/applications
      - install -Dm644 cp data/eu.cichy1173.tabela.metainfo.xml -t /app/share/metainfo
      - install -Dm644 data/icons/hicolor/apps/48x48/eu.cichy1173.tabela.png -t /app/share/icons/hicolor/48x48/apps
      - install -Dm644 data/icons/hicolor/apps/128x128/eu.cichy1173.tabela.png /app/share/icons/hicolor/128x128/apps
```

Comment (hfiguiere) [reply eu.cichy1173.tabela.yml:60]: 128x128 icon was installed in the wrong place.

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

File: eu.cichy1173.tabela.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+runtime: org.kde.Platform
+runtime-version: '6.5'
+sdk: org.kde.Sdk
+command: tabela
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --filesystem=host
```
Comment (hfiguiere) [inline eu.cichy1173.tabela.yml:11]: this is a very wide filesystem permission.

Why is it necessary?

Comment (cichy1173) [reply eu.cichy1173.tabela.yml:11]: Main goal of app is opening csv files for reading content in tables. User also can add new rows and save file as csv or xml.

Comment (cichy1173) [reply eu.cichy1173.tabela.yml]: You are right. It is to much wide. I changed it for home and I removed `network`.

I think, home is the best choice. Why? Main goal of the app is opening users csv files for displaying content in table, and also adding new rows and saving in xml or csv.

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+.flatpak-builder/
+
+build-dir/
```
Comment (travier) [inline .gitignore:4]:
```suggestion
.flatpak-builder/
build-dir/
```

File: eu.cichy1173.tabela.yml [hunk 9]
```diff
@@ -0,0 +1,60 @@
+app-id: eu.cichy1173.tabela
+runtime: org.kde.Platform
+runtime-version: '6.5'
+sdk: org.kde.Sdk
+command: tabela
+finish-args:
+  - --share=ipc
+  - --socket=x11
```
Comment (travier) [inline eu.cichy1173.tabela.yml:8]:
```suggestion
  - --socket=wayland
  - --socket=fallback-x11
```

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

**Comment by cichy1173** [comment]
> Build [58733 successful](https://buildbot.flathub.org/#/builders/6/builds/58733)
> To test this build, install it from the testing repository:

Works fine

**Comment by travier** [comment]
This application should be able to use the file desktop portal to get files from the user home dir instead of relying on static access to all files.

**Comment by cichy1173** [comment]
> This application should be able to use the file desktop portal to get files from the user home dir instead of relying on static access to all files.
> 
> I don't know why this does not work.

I use standard ` QFileDialog.getOpenFileName` from PySide6

**Review by hfiguiere — APPROVED** [review]

**Comment by hfiguiere** [comment]
We could restrict filesystem to `xdg-documents`

**Comment by cichy1173** [comment]
> We could restrict filesystem to `xdg-documents`

I will test this solution locally. If it works fine, I will send new commit.

**Comment by cichy1173** [comment]
bot, build eu.cichy1173.tabela

**Pull request closed without merging**